python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_enumeratefunction.py

```diff
@@ -0,0 +1,204 @@
+# Plot the number of terms in the basis depending on the maximum total degree
+# for several dimension values.
+# We observe the exponential increase of the number of terms with the dimension
+# :math:`d` (curse of dimensionality).
+graph = ot.Graph("Full tensorized basis", "Total degree", "Number of coefficients", True)
+degree_maximum = 10
+list_of_polynomial_degrees = [1, 5, 10, 15, 20]
+point_styles = ["bullet", "circle", "fdiamond", "fsquare", "triangleup"]
+palette = ot.DrawableImplementation.BuildDefaultPalette(len(list_of_polynomial_degrees))
+for i in range(len(list_of_polynomial_degrees)):
+    d = list_of_polynomial_degrees[i]
+    number_of_coeff_array = ot.Sample(degree_maximum, 2)
+    for p in range(1, 1 + degree_maximum):
+        number_of_coeff_array[p - 1, 0] = p
+        number_of_coeff_array[p - 1, 1] = m.comb(p + d, p)
+    cloud = ot.Cloud(number_of_coeff_array)
+    cloud.setPointStyle(point_styles[i])
+    cloud.setLegend(f"$d={d}$")
+    cloud.setColor(palette[i])
+    graph.add(cloud)
+graph.setLegendPosition("topleft")
+graph.setIntegerXTick(True)
+graph.setLogScale(ot.GraphImplementation.LOGY)
+view = otv.View(graph, figure_kw={"figsize": (5, 4)})
+
+# %%
+# Plot the hyperbolic quasi norm for different values of q.
+# With q=1 stratas are hyperplanes, and in case of isotropy
+# it is equivalent to the linear enumeration rule.
+def draw_qnorm(q):
+
+    def qnorm(x):
+        norm = 0.0
+        for xi in x:
+            norm += xi ** q
+        norm = norm ** (1.0 / q)
+        return [norm]
+
+    f = ot.PythonFunction(2, 1, qnorm)
+    f.setInputDescription(['x1', 'x2'])
+    graph = f.draw([0.0] * 2, [1.0] * 2)
+    graph.setTitle(f"q = {q}")
+    return graph
+
+dln = ot.ResourceMap.GetAsUnsignedInteger("Contour-DefaultLevelsNumber")
+ot.ResourceMap.SetAsUnsignedInteger("Contour-DefaultLevelsNumber", 5)
+grid = ot.GridLayout(2, 2)
+grid.setGraph(0, 0, draw_qnorm(1.0))
+grid.setGraph(0, 1, draw_qnorm(0.75))
+grid.setGraph(1, 0, draw_qnorm(0.5))
+grid.setGraph(1, 1, draw_qnorm(0.25))
+ot.ResourceMap.SetAsUnsignedInteger("Contour-DefaultLevelsNumber", dln)
+grid.setTitle("Hyperbolic quasi norm")
+legend_kw={"bbox_to_anchor": (1.0, 1.0), "loc": "upper left"}
+view = otv.View(grid, axes_kw={"aspect": "equal"}, legend_kw=legend_kw)
+
+# %%
+# Plot the multi-indices of the linear enumeration rule by stratas.
+# The lower the value of q the lower the number of interactions terms in stratas.
+grid = ot.GridLayout(2, 2)
+grid.setGraph(0, 0, draw_stratas(ot.HyperbolicAnisotropicEnumerateFunction(dim, 1.0)))
+grid.setGraph(0, 1, draw_stratas(ot.HyperbolicAnisotropicEnumerateFunction(dim, 0.75)))
+grid.setGraph(1, 0, draw_stratas(ot.HyperbolicAnisotropicEnumerateFunction(dim, 0.5)))
+grid.setGraph(1, 1, draw_stratas(ot.HyperbolicAnisotropicEnumerateFunction(dim, 0.25)))
+grid.setTitle("Hyperbolic rule")
+legend_kw={"bbox_to_anchor": (1.0, 1.0), "loc": "upper left"}
```

Comment:
Here it would probably be better to let OT choose the default position for the legends.

Reply:
yes, these are better inside the axes